src/components/LoginOptions/index.js

```diff
@@ -0,0 +1,62 @@
+import PropTypes from 'prop-types';
+import React, {useContext} from 'react';
+import {FormattedMessage} from 'react-intl';
+import Types from 'types';
+
+import {ConfigContext} from 'Context';
+import Literal from 'components/Literal';
+import {getLoginUrl} from 'components/utils';
+
+import LoginOptionsDisplay from './LoginOptionsDisplay';
+
+const LoginOptions = ({form, onFormStart}) => {
+  const config = useContext(ConfigContext);
+
+  const loginAsYourselfOptions = [];
+  const loginAsGemachtigdeOptions = [];
+
+  form.loginOptions.forEach(option => {
+    let readyOption = {...option};
+    readyOption.url = getLoginUrl(option);
+    readyOption.label = (
+      <FormattedMessage
+        description="Login button label"
+        defaultMessage="Login with {provider}"
+        values={{provider: option.label}}
+      />
+    );
+
+    if (readyOption.isForGemachtigde) {
+      loginAsGemachtigdeOptions.push(readyOption);
+    } else {
+      loginAsYourselfOptions.push(readyOption);
+    }
+  });
+
+  if (!form.loginRequired) {
+    loginAsYourselfOptions.push({
+      identifier: 'anonymous',
+      label: <Literal name="beginText" />,
+    });
+  }
+
+  const LoginDisplayComponent = config?.displayComponents?.loginOptions ?? LoginOptionsDisplay;
+  const Container = form.loginRequired ? React.Fragment : 'form';
+  const containerProps = form.loginRequired ? {} : {onSubmit: onFormStart};
```

Comment:
console was complaining about these props after all, so I've broken this up too.